sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsAsyncClient.java

```diff
@@ -0,0 +1,228 @@
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.monitor.query.log.implementation.AzureLogAnalyticsImpl;
+import com.azure.monitor.query.log.implementation.models.BatchRequest;
+import com.azure.monitor.query.log.implementation.models.BatchResponse;
+import com.azure.monitor.query.log.implementation.models.ErrorDetails;
+import com.azure.monitor.query.log.implementation.models.LogQueryRequest;
+import com.azure.monitor.query.log.implementation.models.LogQueryResponse;
+import com.azure.monitor.query.log.implementation.models.QueryBody;
+import com.azure.monitor.query.log.implementation.models.QueryResults;
+import com.azure.monitor.query.log.implementation.models.Table;
+import com.azure.monitor.query.models.ColumnDataType;
+import com.azure.monitor.query.models.LogsQueryBatch;
+import com.azure.monitor.query.models.LogsQueryBatchResult;
+import com.azure.monitor.query.models.LogsQueryBatchResultCollection;
+import com.azure.monitor.query.models.LogsQueryErrorDetails;
+import com.azure.monitor.query.models.LogsQueryOptions;
+import com.azure.monitor.query.models.LogsQueryResult;
+import com.azure.monitor.query.models.LogsTable;
+import com.azure.monitor.query.models.LogsTableCell;
+import com.azure.monitor.query.models.LogsTableColumn;
+import com.azure.monitor.query.models.LogsTableRow;
+import com.azure.monitor.query.models.QueryTimeSpan;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * The asynchronous client for querying Azure Monitor logs.
+ */
+@ServiceClient(builder = LogsClientBuilder.class, isAsync = true)
+public final class LogsAsyncClient {
+
+    private final AzureLogAnalyticsImpl innerClient;
+
+    /**
+     * Constructor that has the inner generated client to make the service call.
+     * @param innerClient The inner generated client.
+     */
+    LogsAsyncClient(AzureLogAnalyticsImpl innerClient) {
+        this.innerClient = innerClient;
+    }
+
+    /**
+     * Returns all the Azure Monitor logs matching the given query in the specified workspaceId.
+     * @param workspaceId The workspaceId where the query should be executed.
+     * @param query The Kusto query to fetch the logs.
+     * @param timeSpan The time period for which the logs should be looked up.
+     * @return The logs matching the query.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<LogsQueryResult> queryLogs(String workspaceId, String query, QueryTimeSpan timeSpan) {
+        return queryLogsWithResponse(new LogsQueryOptions(workspaceId, query, timeSpan))
+                .map(Response::getValue);
```

Comment:
> .map(Response::getValue) [](http://example.com/codeflow?start=15&length=25)

Can LogsQueryResult ever be null\empty? If so, consider using FluxUtil::toMono instead.

Reply:
Same for other use case of it.

Reply:
Nope, LogsQueryResult can never be empty or null.